src/App/main.cpp

```diff
@@ -226,6 +249,10 @@ int main(int argc, char **argv) {
         return EXIT_FAILURE;
     }
 
+    if (!RenderSystem::createShader<ParticleShader>("particle_vert.glsl", "diffuse_frag.glsl", light::dir)) {
+        return EXIT_FAILURE;
```

Comment:
This will cause merge conflicts once my changes are merged. Let's try to merge this PR first.